### PR TITLE
Adding Breathing Space Model and Service

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5567
https://tools.hmcts.net/jira/browse/CIV-5570
https://tools.hmcts.net/jira/browse/CIV-5571
https://tools.hmcts.net/jira/browse/CIV-5573
https://tools.hmcts.net/jira/browse/CIV-5574

### Change description ###
Adding model and common service for the Breathing Space part

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/unit/services/features/breathingSpace/breathingSpaceService.test.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    it('should return an error on redis failure', async () => {
+      mockGetCaseDataFromDraftStore.mockImplementation(async () => {
+        throw new Error(TestMessages.REDIS_FAILURE);
+      });
+
+      await expect(getBreathingSpace('claimId')).rejects.toThrow(TestMessages.REDIS_FAILURE);
+    });
+  });
+
+  describe('saveClaimantResponse', () => {
```
Comment (portizdev) [inline src/test/unit/services/features/breathingSpace/breathingSpaceService.test.ts:83]: Typo error

**Review by portizdev — APPROVED** [review]

File: src/test/unit/services/features/breathingSpace/breathingSpaceService.test.ts (hunk 1 above)
Comment (kenneth-hmcts) [reply src/test/unit/services/features/breathingSpace/breathingSpaceService.test.ts]: Updated

**Review by hmcts-version1-domini — APPROVED** [review]

File: src/main/services/features/breathingSpace/breathingSpaceService.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import {BreathingSpace} from '../../../common/models/breathingSpace';
+import {ClaimDetails} from '../../../common/form/models/claim/details/claimDetails';
+
+const {Logger} = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('interestService');
+
+const getBreathingSpace = async (claimId: string): Promise<BreathingSpace> => {
+  try {
+    const caseData = await getCaseDataFromStore(claimId);
+    return Object.assign(new BreathingSpace(), caseData.claimDetails?.breathingSpace);
```
Comment (lfazelyniaHMCTS) [inline src/main/services/features/breathingSpace/breathingSpaceService.ts:11]: Approved but wanted to mention the below

This is only a preference, but I've been told that Object.assign is more of a hacky/forced way to do things so prefer Maro's way in `getClaimantResponse` where he checks if object exists, and if not, returns new instance of it
`return (claim.claimantResponse) ? claim.claimantResponse : new ClaimantResponse();`
```suggestion
    return (caseData.ClaimDetails?.breathingSpace) ? caseData.claimDetails?.breathingSpace : new BreathingSpace();
```

Comment (kenneth-hmcts) [reply src/main/services/features/breathingSpace/breathingSpaceService.ts]: Changes made, please have a look

Comment (laperla11) [reply src/main/services/features/breathingSpace/breathingSpaceService.ts]: below does the same thing;
`return caseData.ClaimDetails?.breathingSpace ?? new BreathingSpace()`

Comment (kenneth-hmcts) [reply src/main/services/features/breathingSpace/breathingSpaceService.ts]: Updated. Can you have a look?

**Review by laperla11 — APPROVED** [review]

**Pull request merged**